File: nx/test_gmsh.f90
program test_gmsh
    use iso_c_binding
    implicit none
    interface
        subroutine gmsh_initialize(argc, argv) bind(C, name="gmshInitialize")
            import c_int, c_ptr
            integer(c_int), value :: argc
            type(c_ptr), value :: argv
        end subroutine

        function gmsh_is_initialized() bind(C, name="gmshIsInitialized")
            import c_int
            integer(c_int) :: gmsh_is_initialized
        end function

        subroutine gmsh_finalize() bind(C, name="gmshFinalize")
        end subroutine
    end interface

    integer(c_int) :: initialized

    write(*,*) "DEBUG: Testing Gmsh initialization..."
    call gmsh_initialize(0_c_int, c_null_ptr)
    write(*,*) "DEBUG: gmsh_initialize completed."

    initialized = gmsh_is_initialized()
    if (initialized /= 1) then
        write(*,*) "ERROR: Gmsh is not initialized."
    else
        write(*,*) "DEBUG: Gmsh is initialized successfully."
    end if

    call gmsh_finalize()
    write(*,*) "DEBUG: Gmsh finalized."
end program test_gmsh

File: nx/test_gmsh.cpp
#include <gmsh.h>
#include <iostream>

int main(int argc, char **argv) {
    try {
        // Initialize Gmsh
        gmsh::initialize();

        // Log Gmsh version
        std::cout << "Gmsh initialized successfully." << std::endl;
        gmsh::logger::write("Gmsh version logged successfully.", "info");

        // Create a simple model using namespace
        gmsh::model::add("test");

        // Clean up
        gmsh::finalize();

        std::cout << "Test completed successfully" << std::endl;
        return 0;
    }
    catch (const std::exception &e) {
        std::cerr << "Exception caught: " << e.what() << std::endl;
        gmsh::finalize();
        return 1;
    }
}
